base_control/PID.ino

```diff
@@ -0,0 +1,65 @@
+
+PID::PID(){
+  time=millis();
+}
+PID::PID(const double kp, const double ki, const double kd){
+  
+  time=millis();
+
+  this->setTunings(kp,ki,kd);
+
+}
+
+
+void PID::Compute(double setpoint,double &input,double &output,double &resetV){
+  if(millis()-time < sampleTime)
+      return;
+
+  if(resetV){
+    input=(resetV/PULSES_PER_REVOLUTION)*10;
+    resetV=0;
+  }
```

Comment:
What is this for?

Reply:
it calculates rpm at the exact moment, thats why the ticks are passed by parameters. Btw, in the new commit i correct the magic numbers part jeje